balancer-js/src/modules/graph/graph.ts

```diff
@@ -294,7 +302,8 @@ export class PoolGraph {
       linearPool.tokensList[linearPool.mainIndex],
       mainTokenDecimals,
       wrappedTokenNode,
-      proportionOfParent
+      proportionOfParent,
+      upScaledBalances[linearPool.wrappedIndex].toString() // takes price rate into account
```

Comment:
You're right - I made this change in the last minute thinking I'd be ok since I was handling only the wrapped token at this point (which is always 18 decimals) but I'm actually unwrapping it to the mainToken.
I'll update it to be consistent.

Reply:
I updated and double checked it. Scale was indeed wrong before and is correct now. Thanks for catching it!